packages/contracts-bedrock/src/universal/CrossDomainMessenger.sol

```diff
@@ -110,18 +110,23 @@ abstract contract CrossDomainMessenger is
     /// @notice Gas reserved for finalizing the execution of `relayMessage` after the safe call.
     uint64 public constant RELAY_RESERVED_GAS = 40_000;
 
-    /// @notice Gas reserved for message validation within `passesDomainMessageValidator`
-    ///         of `relayMessage`. For L1, this func MUST be a no-op.
-    uint64 public constant L1_RELAY_MESSAGE_VALIDATOR_GAS = 50;
-
-    /// @notice Gas reserved for message validation within `passesDomainMessageValidator`
-    ///         of `relayMessage` in the L2CrossDomainMessenger.
-    uint64 public constant L2_RELAY_MESSAGE_VALIDATOR_GAS = 25_500;
-
     /// @notice Gas reserved for the execution between the `hasMinGas` check and the external
-    ///         call in `relayMessage`.
+    ///         call in `relayMessage`. Does not include `relayMessage` validator config check.
     uint64 public constant RELAY_GAS_CHECK_BUFFER = 5_000;
 
+    /// @notice Gas reserved for the execution of message validation between the `hasMinGas` check
+    ///         the external call in `relayMessage` when the message validation is a no-op (L1).
+    uint64 public constant RELAY_MESSAGE_VALIDATOR_CONFIG_NOOP_GAS = 50;
+
+    /// @notice Gas reserved for the execution of checking the message validator config between the
+    ///         `hasMinGas` check and the external call in `relayMessage` when message validation
+    ///         may occur (L2).
+    uint64 public constant RELAY_MESSAGE_VALIDATOR_CONFIG_CALL_GAS = 6_000;
+
+    /// @notice Gas reserved for the execution of the message validator once the config
+    ///        has been checked and the resulting message validator address is non-zero.
+    uint64 public constant RELAY_MESSAGE_VALIDATOR_GAS = 60_000;
```

Comment:
Not sure if this is just a place holder, but I would reduce this significantly. I think `20_000` would be good for anything we need to do almost all the time.

Reply:
Yeah I didn't put much rigor into the actual value here yet